stdlib/public/core/UnsafePointer.swift.gyb

```diff
@@ -207,7 +207,8 @@ public struct ${Self}<Pointee>
   ///
   /// - Precondition: The `Pointee`s at `self..<self + count` and
   ///   `source..<source + count` are initialized.
-  public func assignFrom(_ source: ${Self}, count: Int) {
+  public func assignFrom(_ source: UnsafePointer<Pointee>, count: Int) {
+    let source = ${Self}(source)
```

Comment:
Oh, I see -- it is for the expression in the precondition to type check.  In that case, I'd prefer to convert `self` into `UnsafePointer`.  This way we won't be creating a mutable pointer to data that is possibly not mutable.